### PR TITLE
Add documentation for MDSWriter, conversion scripts, and supported format

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Add an Introduction to supported dataset format.
- Add an Introduction to MDSWriter
- Add an Introduction to Compression and Hashing algorithms.
- Add a Jupyter Notebook illustrating converting a dataset into MDS format via multiprocess.

## Issue #, if available:
STR-54, STR-55, STR-58

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [x] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [ ] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

**Review by bandish-shah — COMMENTED** [review]
Posting unsubmitted feedback. Still reviewing.

File: docs/source/fundamentals/Introduction_to_writer.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Below is one example where the feature name `x` is an image, and the label `y` is a class value.
+<!--pytest.mark.skip-->
+```python
+column = {
+    'x': 'jpeg',
+    'y': 'int8'
+}
+```
+
+**Advance use-case:** If the data type you are interested in is not listed in the above table, then you can write your own data type class with `encode` and `decode` method in it and patch it inside streaming. For example, you want to write the same for `int32` data type.
```
Comment (knighton) [inline docs/source/fundamentals/Introduction_to_writer.md:38]: Advanced

Comment (karan6181) [reply docs/source/fundamentals/Introduction_to_writer.md]: Fixed in https://github.com/mosaicml/streaming/pull/232/commits/9e77cf4f83e0785e964e0df06f847418aa00364a.

File: docs/source/fundamentals/dataset_format.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Dataset format
```
[the remark this answers is not in the document]
Comment (karan6181) [reply docs/source/fundamentals/dataset_format.md]: Fixed in https://github.com/mosaicml/streaming/pull/232/commits/9e77cf4f83e0785e964e0df06f847418aa00364a.

File: docs/source/fundamentals/Introduction_to_writer.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Dataset conversion guide
+
+If you haven't read the [Introduction to dataset format guide](dataset_format.md), then we highly recommend doing so before you read this.
```
[the remark this answers is not in the document]
Comment (karan6181) [reply docs/source/fundamentals/Introduction_to_writer.md]: Fixed in https://github.com/mosaicml/streaming/pull/232/commits/9e77cf4f83e0785e964e0df06f847418aa00364a.

File: docs/source/fundamentals/Introduction_to_writer.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Dataset conversion guide
```
[the remark this answers is not in the document]
Comment (karan6181) [reply docs/source/fundamentals/Introduction_to_writer.md]: Fixed in https://github.com/mosaicml/streaming/pull/232/commits/9e77cf4f83e0785e964e0df06f847418aa00364a.

**Review by knighton — APPROVED** [review]

**Pull request merged**